### DISCUSSION TITLE
Amazon filters not working

### DISCUSSION THREAD

**Discussion by OldGuyInTheClub** [uBlockOrigin/uAssets#18656] [Q&A] [answered]
### Prerequisites

- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] I am running the latest version of uBlock Origin (uBO).

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

Posting here via https://github.com/uBlockOrigin/uBlock-issues/issues/2492 for support not through Reddit.

I would like to block Amazon results from web searches.  I primarily use Duckduckgo but occasionally use Google as well.  I have found ideas through the comments in:
https://news.ycombinator.com/item?id=29794372
https://news.ycombinator.com/item?id=31693287
and many other web searches.

I have tried the following filters _individually_ and in _various combinations_ but Amazon results keep coming through.
(Full disclosure: I have read the uBlock Origin filter tutorial but frankly don't understand the details of modern webpages enough to know how to write my own.  I use examples from searches and very rarely modify them by trial and error.)

I have used the Element Inspector to see if the hrefs, divs, and other items exist and they appear to be there.  It seems like these filters ought to work but I am no expert.

```adb
duckduckgo.*##.results > div:has(a[href*=["https://www.amazon.com"](https://www.amazon.com/)])

google.com##a[href^=["https://www.amazon."](https://www.amazon./)]:upward(.g)
duckduckgo.*##a[href^=["https://www.amazon."](https://www.amazon./)]:upward(.g)

duckduckgo.com###links > div:has(.result__url__domain:has-text(/amazon/i))
duckduckgo.com###results > div:has(.result__url__domain:has-text(/amazon/i))

duckduckgo.*##.results > div:has(a[href*="amazon.*"])
google.*##.g:has(a[href*="amazon.*"])
```

### URL(s) where the issue occurs.

```
https://wwww.duckduckgo.com
https://www.google.com
```

### uBO version

uBlock Origin 1.50.0

### Browser name and version

Firefox 114.0.2 Mac

### Settings

I have added the filters above to "My Filters", saved the changes, and tried my searches again.
- I have tried them individually 
- I have created a completely new account on my Mac, tried the default uBO configuration, then the added filters one at a time

### Notes

Please see Settings block above

I have also searched the web with multiple search engines for other uBO tips.

**Comment by MasterKia** [comment]
https://letsblock.it/filters/search-results

**Comment by OldGuyInTheClub** [comment]
Did not know that site existed!  Thanks.

It's filters seem to work for Google but not DDG where Amazon results still get through. Should I contact the letsblock.it team for further questions or is it ok to continue here?

```adb
www.google.*##.g:has(a[href*=".amazon."])
www.google.*##a[href*=".amazon."]:upward(1)
duckduckgo.com##a[data-testid="result-title-a"][href*=".amazon."]:upward(.nrn-react-div)
duckduckgo.com##.tile-wrap a[href*=".amazon."]:upward(.tile)
```

**Reply by MasterKia** [reply, marked as answer]
Try this filter:

https://github.com/letsblockit/letsblockit/issues/473#issuecomment-1599649159

**Reply by OldGuyInTheClub** [reply]
That worked!  I appreciate the link to the other thread.  I did some searching before posting but did not come across that one.
